packages/snaps-execution-environments/src/common/endowments/hardenedEndowments.ava.test.ts

```diff
@@ -0,0 +1,290 @@
+  },
+  AbortSignal: {
+    endowments: { AbortSignal },
+    // @ts-expect-error abort() method exists in browser
+    factory: () => AbortSignal.abort(),
+  },
+  DateAttenuated: {
+    endowments: { DateAttenuated },
+    factory: () => new DateAttenuated(),
+  },
+  // --- Objects
+  console: {
+    endowments: { console },
+    factory: () => console,
+  },
+  crypto: {
+    endowments: { crypto },
+    factory: () => crypto,
+  },
+  mathAttenuated: {
+    endowments: { mathAttenuated },
+    factory: () => mathAttenuated,
+  },
+  WebAssembly: {
+    endowments: { WebAssembly },
+    factory: () => WebAssembly,
+  },
+  // --- Functions
+  atob: {
+    endowments: { atob },
+    factory: () => atob('U25hcHM='),
+  },
+  btoa: {
+    endowments: { btoa },
+    factory: () => btoa('Snaps'),
+  },
+  setTimeoutAttenuated: {
+    endowments: { setTimeoutAttenuated },
+    factory: () => setTimeoutAttenuated((param: unknown) => param, 1),
+  },
+  clearTimeoutAttenuated: {
+    endowments: { clearTimeoutAttenuated },
+    factory: () => undefined,
+  },
+  setIntervalAttenuated: {
+    endowments: { setIntervalAttenuated },
+    factory: () => setIntervalAttenuated((param: unknown) => param, 100000),
+  },
+  clearIntervalAttenuated: {
+    endowments: { clearIntervalAttenuated },
+    factory: () => undefined,
+  },
+  fetchAttenuated: {
+    endowments: { fetchAttenuated },
+    factory: () => undefined,
+  },
+};
+
+// These are fake types just to make this test work with the TypeScript
+type HardenedEndowmentSubject = {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  __flag: unknown;
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  prototype: { __flag: unknown };
+};
+type HardenedEndowmentInstance = {
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  __flag: unknown;
+  // eslint-disable-next-line @typescript-eslint/naming-convention
+  __proto__: { __flag: unknown };
+};
+
+/**
+ * Test helper function.
+ *
+ * @param subject - Test subject (instance, object, function).
+ * @param factory - Factory that creates an instance using constructor function.
+ * @returns Array of error messages.
+ */
+function code(
+  subject: HardenedEndowmentSubject,
+  factory: () => HardenedEndowmentInstance,
+): unknown[] {
+  const log = [];
+  const instance = factory();
+  try {
+    subject.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    if (instance) {
+      instance.__flag = 'not_secure';
+    }
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    subject.prototype.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    // eslint-disable-next-line no-proto
+    instance.__proto__.__flag = 'not_secure';
+  } catch (error) {
+    log.push(error.message);
+  }
+  try {
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```

Comment:
If you use `@ts-expect-error`, you can remove the ESLint line above.

Reply:
Changed.